File: Programming and Projection Methods/Lab5C#/Lab5/Lab5/repository/UserRepository.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Lab5.domain;
using log4net;

namespace Lab5.repository
{
    public class UserRepository:IUserRepository
    {
        private static readonly ILog log = LogManager.GetLogger("UserRepository");
        IDictionary<String, String> props;
        public UserRepository(IDictionary<String, String> props)
        {
            log.Info("Creating RideRepository");
            this.props = props;
        }

        public void Delete(string id)
        {
            throw new NotImplementedException();
        }

        public IEnumerable<User> FindAll()
        {
            throw new NotImplementedException();
        }

        public User FindOne(string id)
        {
            log.InfoFormat("Finding User with id {0}", id);
            IDbConnection con = DBUtils.getConnection(props);

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from User where username=@id";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        String username = dataR.GetString(0);
                        String password = dataR.GetString(1);
                        String name = dataR.GetString(2);
                        User user = new User(username,password,name);
                        log.InfoFormat("Exiting findOne with value {0}", user);
                        return user;
                    }
                }
            }
            log.InfoFormat("Exiting findOne with value {0}", null);
            throw new RepositoryException("User inexietent!");
        }

        public User FindOne(string username, string password)
        {
            User u = FindOne(username);
            if (password.Equals(u.Password))
                return u;
            throw new RepositoryException("Parola introdusă este incorectă!");
        }

        public void Save(User e)
        {
            throw new NotImplementedException();
        }

        public void Update(User e)
        {
            throw new NotImplementedException();
        }
    }
}

File: Programming and Projection Methods/Lab2/Tema2/src/main/java/repository/UserRepository.java
package repository;

import domain.User;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import utils.DBConnection;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Properties;

public class UserRepository implements UserRepositoryInterface {
    private static final Logger logger = LogManager.getLogger();
    private DBConnection dbConnection;
    private Validator<User> valid;

    public UserRepository(Properties properties, UserValidator valid) {
        logger.info("Initializing UserRepository with properties: {}", properties);
        this.dbConnection = new DBConnection(properties);
        this.valid = valid;
    }

    @Override
    public User findOne(String usrname) {
        logger.traceEntry("Finding User with id {}", usrname);
        Connection conn = dbConnection.getConnection();
        try (PreparedStatement stat = conn.prepareStatement("select * from User where username=?")) {
            stat.setString(1, usrname);
            try (ResultSet resultSet = stat.executeQuery()) {
                resultSet.next();
                String username = resultSet.getString("username");
                String password = resultSet.getString("password");
                String name = resultSet.getString("name");
                User u = new User(username, password, name);
                return u;
            }
        } catch (SQLException e) {
            logger.error(e);
            throw new RepositoryException("User couldn't be found in the database!");
        }
    }

    @Override
    public User findOne(String username, String password) {
        User u=findOne(username);
        if(password.equals(u.getPassword()))
            return u;
        throw new RepositoryException("Parola introdusă este incorectă!");
    }

    @Override
    public void save(User user) throws ValidationException {

    }

    @Override
    public void update(User user) {

    }

    @Override
    public int size() {
        return 0;
    }

    @Override
    public Iterable<User> findAll() {
        return null;
    }

    @Override
    public void delete(String s) {

    }
}
